### PR TITLE
[AN-5159][Improvements]New adapter for conversation list

### PR DESCRIPTION
**Pull request by JPonte**
New adapter and fragment for the conversation list.
This changes the way archive is accessed and reduce unnecessary redraws of the list.

Tickets: https://wearezeta.atlassian.net/browse/AN-5159

#### APK
[Download build #8838](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8838/artifact/build/artifact/wire-dev-PR847-8838.apk)
[Download build #8839](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8839/artifact/build/artifact/wire-dev-PR847-8839.apk)
[Download build #8840](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8840/artifact/build/artifact/wire-dev-PR847-8840.apk)
[Download build #8841](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8841/artifact/build/artifact/wire-dev-PR847-8841.apk)
[Download build #8844](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8844/artifact/build/artifact/wire-dev-PR847-8844.apk)
[Download build #8851](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8851/artifact/build/artifact/wire-dev-PR847-8851.apk)
[Download build #8853](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8853/artifact/build/artifact/wire-dev-PR847-8853.apk)

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
1. Get an incoming connection request
2. Accept it

AR:
- The newly created 1:1 conversation is at the bottom of the list
- The newly created 1:1 conversation has unread messages counter set to 2

ER:
- The conversation should be at the top of the list
- There should be no unread messages counter there after STR

**Comment by mykola-mokhnach** [comment]
1. Get an incoming connection request when you already have one. The incoming connections list item should be visible

AR:
The count of incoming connection requests is not changed. It's only changed when you scroll the corresponding item out of the viewport and back

ER:
The count of pending requests should  be properly updated after STR

**Comment by mykola-mokhnach** [comment]
Swipe to the right from the list of archived conversations should not open conversation view

**Comment by mykola-mokhnach** [comment]
1. Open the list of archived  conversations
2. tap any conversation to unarchive it
3. Tap Back button

AR:
Archive view freezes. Close button does not work anymore

ER:
Conversations list should be shown. Archive view should be closed after

**Comment by mykola-mokhnach** [comment]
1. Create a brand new user w/o conversations
2. Receive an incoming connection request

AR:
Conversations list is empty after STR, although I see the pending connection in search

ER:
Pending connection item should be visible in conversations list after STR

**Comment by mykola-mokhnach** [comment]
Archived pending outgoing connections are not visible in archive

File: app/src/main/scala/com/waz/zclient/views/conversationlist/ConversationListRow.scala [hunk 1]
```diff
@@ -196,53 +197,45 @@ class ConversationListRow(context: Context, attrs: AttributeSet, style: Int) ext
...
-          conversationId.publish(Some(ConvId(iConversation.getId)), Threading.Background)
-      }
+      avatar.setAlpha(1f)
+      conversationId.publish(Some(conversationData.id), Threading.Background)
       closeImmediate()
     }
   }
 
+  def setConversation(iConversation: IConversation): Unit = {
+    //TODO: remove
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/views/conversationlist/ConversationListRow.scala]: remove?

**Comment by mykola-mokhnach** [comment]
Archive icon should not be visible if there are no archived conversations

**Comment by mykola-mokhnach** [comment]
1. Send a connection request to userA
2. Find this pending request in the list

AR:
Unique username is not present in pending conversation subtitle. It is necessary tor estart Wire to see it

ER:
Unique username of userA should be visible as subtitle immediately after STR

**Comment by mykola-mokhnach** [comment]
Archive is not closed automatically if the only item there was pending outgoing connection

**Comment by mykola-mokhnach** [comment]
https://wearezeta.atlassian.net/browse/AN-5220
https://wearezeta.atlassian.net/browse/AN-5219
https://wearezeta.atlassian.net/browse/AN-5218

**Comment by JPonte** [comment]
build again

**Pull request merged**